db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/provision.rb

```diff
@@ -1 +1,96 @@
+def get_custom_tag(type)
+  type + "_#{$evm.root['automation_task'][:id]}"
+end
 
+def create_custom_tag(name)
+  tag_name = "#{name}_#{$evm.root['automation_task'][:id]}"
+  unless $evm.execute('tag_exists?', 'deploy', tag_name)
+    $evm.log(:info, "********************** creating tag ***************************")
+    $evm.execute('tag_create',
+                 'deploy',
+                 :name        => tag_name,
+                 :description => tag_name)
+  end
+end
+
+def tagged_tasks
+  tasks = []
+  $evm.vmdb('miq_provision_request').all.each do |prov|
+    next unless prov.get_tags[:deploy] &&
+      (prov.get_tags[:deploy].include?(get_custom_tag("master")) ||
+        prov.get_tags[:deploy].include?(get_custom_tag("node")))
+    tasks << prov
+  end
+  tasks
+end
+
+def create_provision_requests(vm_base_name, num_of_vms, args, type)
+  (1..num_of_vms).each do |num|
+    args[2]['vm_name'] = vm_base_name + "_" + num.to_s
+    request            = $evm.execute('create_provision_request', *args)
+    request.add_tag("deploy", type + '_' + $evm.root['automation_task'][:id].to_s)
+  end
+end
+
+def check_state(number_of_finished_vms, tasks, fail)
+  if number_of_finished_vms == tasks.count && tasks.count > 0
+    $evm.root['ae_result'] = "ok"
```

Comment:
@mkanoor For now yea, but your right it would be better to remove them, can you send me an short snippet on how to spawn a request that will remove a vm?